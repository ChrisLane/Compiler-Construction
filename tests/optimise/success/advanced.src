main() {
    var x = 100;
    var y = 200;
    var z = false;

    var a = false;

    a =
        if ($x <= $y) {
            !($x == $y) && !$z;
        } else {
            $a && true;
        };
    print $a;
}
