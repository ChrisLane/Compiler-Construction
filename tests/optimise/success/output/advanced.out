
main (  ) { 
    New ( "x" = Const 100 ) In { 
        New ( "y" = Const 200 ) In { 
            New ( "z" = Bool false ) In { 
                New ( "a" = Bool true ) In { 
                    Print ( Bool true ) 
                } 
            } 
        } 
    } 
}

